### ISSUE TITLE
[Question] Components not updating in time when using React

### ISSUE THREAD

**Issue by antpingelli** [microsoft/playwright#13740]
I have a UI that allows you to create a certain resource and saves the record in our DB. On success, the UI then redirects to a View page where that record is fetched on page load and displayed back to the user. I have tests for this flow but have been noticing intermittent failures saying that one of the fields I check is empty, even though I can validate that it does have data _eventually_. I believe what is happening is since we are using Redux, the fetch finishes but the components have not yet updated, and so the test validates just before.

My current approach is 
```
// trigger save and validate request body
let [response] = await Promise.all([page.waitForResponse(reqRegex), page.click(selectors.saveButton)]);
const requestData = JSON.parse(response.request().postData());
expect(requestData).toEqual(expectedBody);
// wait for page spinner to be hidden, which happens when the FETCH_SUCCESS action is dispatched
await page.waitForSelector(selectors.spinner, { state: 'hidden' });
// validate name on View page
await expect(page.locator(selectors.nameField)).toHaveValue(name);
```

I have thought about change the validation check to be `page.waitForSelector` since I think that would allow for the timing discrepancy but that loses the `expect` output of showing what value we expected vs received. Is there another process I can try or am I just thinking about testing React incorrectly?

Thanks!

**Comment by pavelfeldman** [comment]
I don't see anything react-specific in this snippet. Following should just work for you.

```js
await page.click(selectors.saveButton);
await expect(page.locator(selectors.nameField)).toHaveValue(name);
```

In the snippet above, Playwright will wait for the name field to have a given value, so _eventually_ should be covered by this snippet. Your additional assertions for response and data there all look good. The only thing I'm not sure you need is for this spinner to be hidden.

**Comment by antpingelli** [comment]
I mentioned React since I think that is the source of this issue, but the code I provided is just playwright so that does not involve React directly.

> Playwright will wait for the name field to have a given value

I was able to add a small delay after the fetch returns but before the component updates and am consistently seeing it fail. If I change the validation to `await page.waitForSelector('#name-field[value="NAME"]');` it works. I do not see anywhere in the docs that `page.locator` waits, I was under the impression that it immediately gets the value
